utils/src/main/scala/com.gu.contentatom.renderer/utils/CapiService.scala

```diff
@@ -0,0 +1,129 @@
+package com.gu.contentatom.renderer.utils
+
+import com.gu.contentapi.client.{ContentApiClient, GuardianContentClient}
+import com.gu.contentapi.client.model.v1.ItemResponse
+import com.gu.contentatom.renderer._
+import com.gu.contentatom.thrift.AtomType
+import cats.data.OptionT
+import cats.effect._
+import cats.implicits._
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.implicits._
+import org.http4s.server.Router
+import org.http4s.server.blaze._
+import org.http4s.syntax._
+import org.http4s.twirl._
+import scala.concurrent.ExecutionContext.Implicits.global
+import play.twirl.api.Html
+
+
+object Main extends IOApp {
+  object AtomIdMatcher extends QueryParamDecoderMatcher[String]("id")
+
+  implicit val atomTypeParamDecoder: QueryParamDecoder[AtomType] = new QueryParamDecoder[AtomType] {
+    def decode(value: QueryParameterValue) = value.value match {
+      case "audio"    => AtomType.Audio.valid
+      case "chart"    => AtomType.Chart.valid
+      case "guide"    => AtomType.Guide.valid
+      case "profile"  => AtomType.Profile.valid
+      case "qanda"    => AtomType.Qanda.valid
+      case "timeline" => AtomType.Timeline.valid
+      case _          => ParseFailure(value.value, "Invalid atom type").invalidNel
+    }
+  }
+
+  object AtomTypeMatcher extends QueryParamDecoderMatcher[AtomType]("at")
+
+  val config = ArticleConfiguration(
+    ajaxUrl = "",
+    audioSettings = AudioSettings(false),
+    commonsdivisionConfiguration = ArticleConfiguration.CommonsdivisionConfiguration(true)
+  )
+
+  val client = new GuardianContentClient(sys.env("CAPI_TEST_KEY"))
+
+  val assetsService: HttpRoutes[IO] = HttpRoutes.of[IO] {
+    case req @ GET -> Root / "styles" ~ "css" =>
+      StaticFile.fromResource("/styles.css", global, Some(req)).getOrElseF(NotFound())
+  }
+
+  val capiService: HttpRoutes[IO] = HttpRoutes[IO] {
+    case GET -> Root :? AtomTypeMatcher(atomType) +& AtomIdMatcher(atomId) =>
```

Comment:
Hmm I find Scala syntax like this very hard to understand. Perhaps you can talk me through this on Monday?

Reply:
Yes of course